pkg/services/network/nsxt_vpc_provider.go

```diff
@@ -0,0 +1,186 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	nsxopv1 "github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha1"
+	vmoprv1 "github.com/vmware-tanzu/vm-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/util/conditions"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	vmwarev1 "sigs.k8s.io/cluster-api-provider-vsphere/apis/vmware/v1beta1"
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/context/vmware"
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/services"
+)
+
+// nsxtVPCNetworkProvider provisions nsx-vpc type cluster network.
+type nsxtVPCNetworkProvider struct {
+	client client.Client
+}
+
+// NSXTVpcNetworkProvider returns an instance of nsx-vpc type network provider.
+func NSXTVpcNetworkProvider(client client.Client) services.NetworkProvider {
+	return &nsxtVPCNetworkProvider{
+		client: client,
+	}
+}
+
+func (vp *nsxtVPCNetworkProvider) HasLoadBalancer() bool {
+	return true
+}
+
+func (vp *nsxtVPCNetworkProvider) SupportsVMReadinessProbe() bool {
+	// Note: The control plane VM network is private with nsx-vpc and
+	// readiness probe would fail. Therefore, nsxvpcNetworkProvider
+	// doesn't support VM readiness probe.
+	return false
+}
+
+// verifyNsxtVpcSubnetSetStatus checks the status conditions of a given SubnetSet within a cluster context.
+// If the subnet isn't ready, it is marked as false, and the function returns an error.
+// If the subnet is ready, the function updates the VSphereCluster with a "true" status and returns nil.
+func (vp *nsxtVPCNetworkProvider) verifyNsxtVpcSubnetSetStatus(vspherecluster *vmwarev1.VSphereCluster, subnetset *nsxopv1.SubnetSet) error {
+	clusterName := vspherecluster.Name
+	namespace := vspherecluster.Namespace
+	hasReadyCondition := false
+
+	for _, condition := range subnetset.Status.Conditions {
+		if condition.Type == nsxopv1.Ready {
+			hasReadyCondition = true
+			if condition.Status != corev1.ConditionTrue {
+				conditions.MarkFalse(vspherecluster, vmwarev1.ClusterNetworkReadyCondition, vmwarev1.ClusterNetworkProvisionFailedReason, clusterv1.ConditionSeverityWarning, condition.Message)
+				return errors.Errorf("subnetset ready status is: '%s' in cluster %s. reason: %s, message: %s",
+					condition.Status, types.NamespacedName{Namespace: namespace, Name: clusterName}, condition.Reason, condition.Message)
+			}
+		}
```

Comment:
Makes it a bit more readable to me :-)

```suggestion
		if condition.Type != nsxopv1.Ready {
			continue
		}
		hasReadyCondition = true
		if condition.Status != corev1.ConditionTrue {
			conditions.MarkFalse(vspherecluster, vmwarev1.ClusterNetworkReadyCondition, vmwarev1.ClusterNetworkProvisionFailedReason, clusterv1.ConditionSeverityWarning, condition.Message)
			return errors.Errorf("subnetset ready status is: '%s' in cluster %s. reason: %s, message: %s",
				condition.Status, types.NamespacedName{Namespace: namespace, Name: clusterName}, condition.Reason, condition.Message)
		}
```

Reply:
Sure. The use continue could make the control flow more clear.